pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -196,7 +198,13 @@ func (p *PollerWorker) Poll(execAgentCommands agentCommandsExecutor, closeChan c
 	// do the first round to fill the cache immediately
 	execAgentCommands(p.AgentCommands)
 
-	ticker := time.NewTicker(time.Second * p.CallTick)
+	pollStableInterval := p.CallTick * time.Second
+	pollInterval := pollStableInterval
+
+	if pollInitialInterval < pollStableInterval {
+		pollInterval = pollInitialInterval
+	}
+	ticker := time.NewTicker(pollInterval)
```

Comment:
Deserver a function maybe ?
creates a backoff ticker

Reply:
If you have a better idea, please show how it will be used to convince that it is better.

Reply:
1. Just create a wrapper function to this 6 lines, because they do one thing - create a timer.
2. We can create, its up to you, a wrapper struct with methods that provides this initialize + the incrementPollInterval, replaceTicker methods.

Reply:
"show how" means exactly that.. give a usage snippet and convince me it is better. At the moment I cannot imagine your thoughts of how this will look like.

These lines you commented here are directly related to the ones if the for-loop, so this is not one thing, it is a half thing. If your usage example will look better, then I'll consider it.

Reply:
https://flaviocopes.com/golang-is-go-object-oriented/#:~:text=Yes%20and%20no.,there%20is%20no%20type%20hierarchy.&text=Moreover%2C%20methods%20in%20Go%20are,plain%2C%20%E2%80%9Cunboxed%E2%80%9D%20integers.

Struct embedding that will implement a wrapper to the 3 methods of the backoff ticker

Reply:
How is this link helpful in showing me how you want to see something used in this implementation?
I asked to show me how you want this to look like here, not for a learning session on the web from some blogger.

If you have something explicit to present/suggest, please do so. Otherwise, I see no action item here.

Reply:
See please the section `Struct embedding` there
which shows a struct with methods, i dont think there is more needed to elaborate about it (this is my opinion in this case),
you created a new type - backoff timer, it has 3 functions,
if you want you can create a struct with methods that will Implement it OO style kinda.

Reply:
I did not ask to quote the blog, nor for a Golang OO programming short course.
I asked to use an example of how you like it to look on the code in this PR and not a general Go structure.

Reply:
From my view, this is the deepest i can / decide to dedicate atm, we can take it to other PR if desired, thanks